### PR TITLE
Support preconfigured cache with eternal="true"

### PR DESCRIPTION
**Pull request by jdillon**
To resolve https://github.com/ehcache/ehcache-jcache/issues/40

### CODE REVIEW

**Comment by terracotta-org** [comment]
Can one of the admins verify this patch?

File: ehcache-jcache/src/main/java/org/ehcache/jcache/JCacheConfiguration.java [hunk 1]
```diff
@@ -36,6 +34,8 @@
 import javax.cache.integration.CacheLoader;
 import javax.cache.integration.CacheWriter;
 
+import net.sf.ehcache.config.CacheConfiguration;
```
Comment (jdillon) [inline ehcache-jcache/src/main/java/org/ehcache/jcache/JCacheConfiguration.java]: sorry about the import change, forgot to flip that IDE automagic off when committing this.

**Comment by jdillon** [comment]
Anyone?

**Comment by jdillon** [comment]
@alexsnaps signed agreement is in the email

**Comment by alexsnaps** [comment]
@jdillon In order to accept the PR, we'd need a test for the change... Can you work on adding this? 
Also, besides the `import` line being moved, there are a couple of inconsistencies wrt code formatting. 
Let me know if doing the test works, otherwise I'll go about all of the above myself. Thanks

**Comment by jdillon** [comment]
@alexsnaps simple test added.

BTW I noticed this in the pom.xml:

```
        <!--With slf4j, users choose a concrete logging implementation at deploy time-->
        <dependency>
            <groupId>log4j</groupId>
            <artifactId>log4j</artifactId>
            <version>1.2.16</version>
            <scope>test</scope>
        </dependency>
```

This does not actually provide a slf4j-impl, and since the tests do not seem to even be configuring log4j, you are probably better off using:

```
        <dependency>
            <groupId>org.slf4j</groupId>
            <artifactId>slf4j-simple</artifactId>
            <version>1.6.1</version>
            <scope>test</scope>
        </dependency>
```

**Comment by jdillon** [comment]
@alexsnaps unsure which style/format stuff you want changed, assumed you guys would just reformat as you need.  Its hard to juggle everyones style ;-)

File: ehcache-jcache/src/main/java/org/ehcache/jcache/JCacheConfiguration.java (hunk 1 above)
Comment (alexsnaps) [inline ehcache-jcache/src/main/java/org/ehcache/jcache/JCacheConfiguration.java]: If you could remove it up, that'd be appreciated. Along with...

File: ehcache-jcache/src/main/java/org/ehcache/jcache/JCacheConfiguration.java [hunk 2]
```diff
@@ -93,22 +93,28 @@ public JCacheConfiguration(final CacheConfiguration cacheConfiguration, final Co
...
-
-                    @Override
-                    public Duration getExpiryForUpdate() {
-                        return getExpiryForCreation();
-                    }
-                };
+                if (cacheConfiguration.isEternal()) {
+                    expiryPolicy = EternalExpiryPolicy.factoryOf().create();
+                }
+                else {
```
Comment (alexsnaps) [inline ehcache-jcache/src/main/java/org/ehcache/jcache/JCacheConfiguration.java]: ... this small style fix here, i.e. 

```
} else {
```

**Comment by jdillon** [comment]
@alexsnaps done

**Comment by alexsnaps** [comment]
I understand why the request sounded weird now... I'll do it, but I'll squash that last commit, fairly confident this will keep you as the author et al.

**Comment by jdillon** [comment]
@alexsnaps i'm less concerned about being retained as author as I am in getting that bug fixed :-P

Did you see my comment about the pom?  I didn't want to leak that change into this PR, but asis the logging configuration for slf4j is busted/non-functional for scope=test.

**Pull request closed without merging**